### ISSUE TITLE
모델 불변성 검사

### ISSUE THREAD

**Issue by gyuwon** [envicase/ReactiveMvvm#31]
Reactive-MVVM의 모델 철학중 하나가 불변성입니다. 그래서 프레임워크 차원에서 모델의 불변성을 검사하는 논리가 들어가면 좋을 것 같은데 리플렉션 권한 등의 이슈가 있을 것 같습니다. 좋은 방법 없을까요?

**Comment by jwChung** [comment]
@gyuwon 저도 언뜻 떠오르는 생각은 reflection을 이용해 아래와 같이 불변성을 해치는 멤버를 테스트하는 것이라 생각하는데요. 리플렉션 권한을 잘 몰라서 이게 public API에도 권한이 제한되는지 모르겠네요.

* public field가 있는지
* public setter가 있는지 
* `Array`와 같이 내부 데이타를 변경할 수 있는 타입을 반환하는 속성이 있는지

**Comment by gyuwon** [comment]
@jwChung 아 그렇네요. 공용 멤버만 다루면 되니 권한문제가 발생하진 않겠네요. 불변성 검사 논리를 @jwChung 님이 맡아주시면 좋을 것 같네요. 현재 제가 하고있는 작업과 독립적으로 진행할 수도 있겠고요. 그리고 혹시 가능하다면 작업을 나눠서 코딩 테스트에 사용해도 될 듯 합니다.

**Comment by jwChung** [comment]
네 확인했습니다. 제가 맡아서 진행해보겠습니다. 코딩테스트에 대해서는 따로 얘기 한번 나누시죠. 저도 제 의견을 말씀 드리고 싶은 부분도 있고, @gyuwon 님 의견도 듣고싶네요.
